cpp/src/io/json/host_tree_algorithms.cu

```diff
@@ -921,9 +945,32 @@ void scatter_offsets(tree_meta_t const& tree,
              column_categories[col_ids[parent_node_id]] == NC_LIST and
              (!d_ignore_vals[col_ids[parent_node_id]]);
     });
+  // For children of list and in ignore_vals, find it's parent node id, and set corresponding
+  // parent's null mask to null. Setting mixed type list rows to null.
+  thrust::for_each_n(
+    rmm::exec_policy_nosync(stream),
+    thrust::make_counting_iterator<size_type>(0),
+    list_children_end - thrust::make_zip_iterator(node_ids.begin(), parent_col_ids.begin()),
```

Comment:
Should this use `thrust::distance` ?
```suggestion
    thrust::distance(thrust::make_zip_iterator(node_ids.begin(), parent_col_ids.begin(), list_children_end),
```

Reply:
The same expression is used to compute `num_list_children` below this line.

Reply:
Should be something like this:
```
auto const list_children_begin = thrust::make_zip_iterator(node_ids.begin(), parent_col_ids.begin();
// replace all `thrust::make_zip_iterator` by `list_children_begin`.
auto const num_list_children = thrust::distance(list_children_begin, list_children_end);
// use num_list_children
```